realm/src/main/java/io/realm/Realm.java

```diff
@@ -1206,7 +1206,12 @@ public void addChangeListener(RealmChangeListener listener) {
      */
     public void removeChangeListener(RealmChangeListener listener) {
         checkIfValid();
-        changeListeners.remove(new WeakReference<RealmChangeListener>(listener));
+        Iterator<WeakReference<RealmChangeListener>> iterator = changeListeners.iterator();
+        while (iterator.hasNext()) {
```

Comment:
So I guess making a defensive copy of the list of weak references is still a good idea